docs/pages/blog/mui-x-v5.md

```diff
@@ -0,0 +1,190 @@
+This can be seen in the top part of the screenshot below, where it compares `v5.0.0-beta.4` (the last version before the new virtualization) with `v5.0.0`.
+On each scroll event, it renders again and each frame takes a long time to be drawn (some are even lost).
+To address these problems, we took the following actions:
+
+- Avoid rendering the entire grid during scroll
+- Pass the correct value to the `key` prop to ensure that React will reuse the existing DOM nodes
+- Reduce the number of event listeners attached to each cell
+- Increase the number of columns rendered in the overscan (the extra columns rendered to make scroll smoother)
+- Delaying the rendering of new columns whenever it was possible
+
+The result of these actions is on the bottom part of the comparison.
+The number of frames that could be drawn in the same amount of time dramatically increased, compared to the previous virtualization.
+In our [benchmark](https://github.com/mui-org/material-ui-x/pull/2673), the FPS (frames per second) went from 22 to 42, on average.
+The time each frame takes to be rendered, indicated by the width of each block, was reduced.
+The idea of delaying the re-rendering also can be seen in the large voids between the blocks.
+Each void means that a re-render was not necessary since the required columns were already rendered by the overscan.
+
+![Performance comparison between v5.0.0-beta.4 and v5.0.0](https://user-images.githubusercontent.com/42154031/141475697-11281f83-9a2f-4a2b-8001-f459cbdccbd8.png)
+
+Some of the mentioned improvements were also applied to the rows, however, the gains were more subtle.
+Besides the better performance, the new virtualization also brings the following fixes:
+
+- Horizontal and vertical scroll share the same logic
+- No more jumps when changing the rendered rows
+- Calling `apiRef.current.scrollToIndexes` works no matter where the cell is
+- Improved support for when virtualization is disabled
+- Fix keyboard navigation with arrow keys.
+
+## Improved state management
+
+Several improvements were made to our state management to improve developer experience, performances and consistency in the execution order.
+
+#### Improve the developer experience around the state
+
+To improve developer experience for both the X-team and developers using the `apiRef` features, we are working on making the state structure and the tools to access it as easy to understand as possible.
+
+- We removed the `state` structure from the public API. Access data in the state should always be done through `apiRef` methods (`apiRef.current.getSelectedRows`) or selectors (`selectedGridRowsSelector`).
+- We renamed most selectors to have a consistent naming convention
+- We restructured our state so that each feature has a single sub state and is the only one to update it (`state.filter` is only managed by our `useGridFilter` hooks which exposes methods for both internal and 3rd party code to interact with this state).
```

Comment:
```suggestion
- We restructured our state so that each feature has a single substate, and is the only one to update it (`state.filter` is only managed by our `useGridFilter` hooks which exposes methods for both internal and 3rd party code to interact with this state).
```